androidApp/src/main/java/com/mbta/tid/mbta_app/android/component/PredictionRowView.kt

```diff
@@ -52,34 +51,37 @@ fun PredictionRowView(
     destination: @Composable () -> Unit
 ) {
     Row(
-        modifier
-            .fillMaxWidth()
-            .heightIn(min = 44.dp)
-            .padding(8.dp)
-            .background(color = MaterialTheme.colorScheme.background),
+        modifier.fillMaxWidth(),
         horizontalArrangement = Arrangement.SpaceBetween,
         verticalAlignment = Alignment.CenterVertically,
     ) {
+        if (pillDecoration is PillDecoration.OnRow) {
+            RoutePill(
+                pillDecoration.route,
+                line = null,
+                RoutePillType.Flex,
+                modifier =
+                    if (predictions.secondaryAlert == null) Modifier.padding(end = 8.dp)
+                    else Modifier
+            )
+        }
         predictions.secondaryAlert?.let { secondaryAlert ->
             Image(
                 painterResource(drawableByName(secondaryAlert.iconName)),
                 stringResource(R.string.alert),
-                modifier = Modifier.placeholderIfLoading()
+                modifier = Modifier.placeholderIfLoading().padding(end = 8.dp)
             )
         }
-        if (pillDecoration is PillDecoration.OnRow) {
-            RoutePill(pillDecoration.route, line = null, RoutePillType.Flex)
-        }
 
-        Column(modifier = Modifier.weight(1f).padding(start = 8.dp)) { destination() }
+        Column(modifier = Modifier.weight(1f)) { destination() }
         Row(
-            modifier = Modifier.weight(1f),
+            modifier = Modifier.width(IntrinsicSize.Max),
```

Comment:
I’ve asked [in Slack](https://mbta.slack.com/archives/C05QMG9GS9M/p1744304603018869) whether this change is worth keeping.

Reply:
We settled on a different tradeoff instead.